src/Raven.Studio/typescript/viewmodels/resources/createDatabase.ts

```diff
@@ -242,20 +245,106 @@ class createDatabase extends dialogViewModelBase {
             return "unchecked";
         });
 
-
-        this.databaseLocationShowing = ko.pureComputed(() => {
-            return this.databaseLocationCalculated();
-        });
-
         this.databaseModel.path.dataPath.throttle(300).subscribe((newPathValue) => {
             if (this.databaseModel.path.dataPath.isValid()) {
-                new getDatabaseLocationCommand(this.databaseModel.name(), newPathValue)
-                    .execute()
-                    .done((fullPath: string) => this.databaseLocationCalculated(fullPath))
+                this.updateDatabaseLocationInfo(this.databaseModel.name(), newPathValue);
+                this.updateFolderPathOptions(newPathValue);
             } else {
-                this.databaseLocationCalculated("Invalid path");
+                this.databaseLocationInfo([]);
+                this.spinners.databaseLocationInfoLoading(false);
             }
         });
+
+        this.databaseLocationInfoToDisplay = ko.pureComputed(() => {
+            const databaseLocationInfo = this.databaseLocationInfo();
+            const selectedNodes = this.databaseModel.replication.nodes().map(x => x.tag());
+            const replicationFactor = this.databaseModel.replication.replicationFactor();
+            if (this.databaseModel.replication.manualMode()) {
+                if (selectedNodes.length === 0) {
+                    return [];
+                }
+            }
+            else if (replicationFactor === 0) {
+                return [];
+            }
+
+            return databaseLocationInfo.filter(x => selectedNodes.indexOf(x.NodeTag) > -1);
+        });
+
+        this.databaseLocationInfoMessage = ko.pureComputed(() => {
+            const message = "The data files will be created in";
+            const selectedNodes = this.databaseModel.replication.nodes().map(x => x.tag());
+            if (this.databaseModel.replication.manualMode()) {
+                if (selectedNodes.length === 0) {
+                    return null;
+                }
+
+                return `${message}:`;
+            }
+
+            const numberOfClusterNodes = this.clusterNodes.slice().length;
+            const replicationFactor = Math.min(this.databaseModel.replication.replicationFactor(), numberOfClusterNodes);
+            if (replicationFactor === 0) {
+                return null;
+            }
+
+            if (replicationFactor === 1 ||
+                replicationFactor === numberOfClusterNodes) {
+                return `${message}:`;
+            }
+
+            return `${message} ${replicationFactor} of the following nodes:`;
+        });
+
+        this.allAutoCompleteOptions = ko.pureComputed(() => {
+            const result: { path: string, isRecent: boolean }[] = [];
```

Comment:
I wonder if we need 2 separate groups here: recent and not-recent - I know recent are bold on UI. I think we should have single style for both and make sure we don't have duplicates on this list.

Reply:
however .... recent are sorted by usage date and other are sorted A-Z, so maybe we should leave it.